core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java

```diff
@@ -164,15 +199,31 @@ public boolean equals(Object o)
       return false;
     }
     CloudObjectInputSource that = (CloudObjectInputSource) o;
-    return Objects.equals(uris, that.uris) &&
+    return Objects.equals(scheme, that.scheme) &&
+           Objects.equals(uris, that.uris) &&
            Objects.equals(prefixes, that.prefixes) &&
-           Objects.equals(objects, that.objects);
+           Objects.equals(objects, that.objects) &&
+           Objects.equals(filter, that.filter);
   }
 
   @Override
   public int hashCode()
   {
-    return Objects.hash(uris, prefixes, objects);
+    return Objects.hash(scheme, uris, prefixes, objects, filter);
+  }
+
+  private void illegalArgsChecker() throws IllegalArgumentException
+  {
```

Comment:
Please add some validation to filter as its user input.

Reply:
what sort of validation do you expect @cryptoe? `filter` is nullable.

Reply:
I was thinking of a valid filter check like a valid regex, but seems like we do not have that helper method here: 
http://commons.apache.org/proper/commons-io/apidocs/org/apache/commons/io/filefilter/WildcardFileFilter